File: test-service-java/src/main/java/com/example/testservicejava/v1/controllers/UserController.java
package com.example.testservicejava.v1.controllers;

import com.example.testservicejava.v1.entities.User;
import com.example.testservicejava.v1.services.UserService;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(path = "/users")
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
    public ResponseEntity<List<User>> readAllUsers() {
        return ResponseEntity.ok(userService.readAllUsers());
    }
}

File: test-service-kotlin/src/main/kotlin/org/xapps/service/testservice/v1/controllers/UserController.kt
package org.xapps.service.testservice.v1.controllers

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.xapps.service.testservice.v1.entities.User
import org.xapps.service.testservice.v1.services.UserService

@RestController
@RequestMapping(path = ["/users"])
class UserController @Autowired constructor(
    private val userService: UserService
) {

    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
    fun readAllUsers(): ResponseEntity<List<User>> =
        ResponseEntity.ok(userService.readAllUsers())

}